test/trento/heartbeats_test.exs

```diff
@@ -17,131 +18,141 @@ defmodule Trento.HeartbeatsTest do
+        :utc_now,
+        fn -> updated_time end
       )
 
-    expect(
-      Trento.Support.DateService.Mock,
-      :utc_now,
-      fn -> updated_time end
-    )
+      scenario_setup(@scenario, agent_id, :passing)
 
-    expect(
-      Trento.Commanded.Mock,
-      :dispatch,
-      fn command ->
-        assert %UpdateHeartbeat{
-                 host_id: ^agent_id,
-                 heartbeat: :passing
-               } = command
+      Heartbeats.heartbeat(agent_id, Trento.Support.DateService.Mock)
 
-        :ok
-      end
-    )
+      [heartbeat] = Repo.all(Heartbeat)
 
-    Heartbeats.heartbeat(agent_id, Trento.Support.DateService.Mock)
+      assert heartbeat.timestamp == updated_time
+    end
 
-    [heartbeat] = Repo.all(Heartbeat)
+    test "dispatch commands on heartbeat expiration scenario:#{@scenario}" do
+      %{id: agent_id} = insert(:host, heartbeat: :passing)
+      %{timestamp: now} = insert(:heartbeat, agent_id: agent_id)
 
-    assert heartbeat.timestamp == updated_time
-  end
+      expired_time =
+        DateTime.add(
+          now,
+          Application.get_env(:trento, Heartbeats)[:interval] + 1,
+          :millisecond
+        )
 
-  test "dispatch commands on heartbeat expiration" do
-    %{id: agent_id} = insert(:host, heartbeat: :passing)
-    %{timestamp: now} = insert(:heartbeat, agent_id: agent_id)
+      expect(
+        Trento.Support.DateService.Mock,
+        :utc_now,
+        fn -> expired_time end
+      )
 
-    expired_time =
-      DateTime.add(
-        now,
-        Application.get_env(:trento, Heartbeats)[:interval] + 1,
-        :millisecond
+      scenario_setup(@scenario, agent_id, :critical)
+      Heartbeats.dispatch_heartbeat_failed_commands(Trento.Support.DateService.Mock)
+    end
+
+    test "filter deregistered hosts from heartbeat failed check scenario: #{@scenario}" do
+      %{id: agent_id} = insert(:host, heartbeat: :passing, deregistered_at: DateTime.utc_now())
+      %{timestamp: now} = insert(:heartbeat, agent_id: agent_id)
+
+      expired_time =
+        DateTime.add(
+          now,
+          Application.get_env(:trento, Heartbeats)[:interval] + 1,
+          :millisecond
+        )
+
+      expect(
+        Trento.Support.DateService.Mock,
+        :utc_now,
+        fn -> expired_time end
       )
 
-    expect(
-      Trento.Support.DateService.Mock,
-      :utc_now,
-      fn -> expired_time end
-    )
+      expect(
+        Trento.Commanded.Mock,
+        :dispatch,
+        0,
+        fn _ -> :ok end
+      )
+
+      Heartbeats.dispatch_heartbeat_failed_commands(Trento.Support.DateService.Mock)
+    end
+  end
 
+  defp scenario_setup(:with_correlation, agent_id, health) do
     expect(
       Trento.Commanded.Mock,
       :dispatch,
-      fn command ->
-        assert %UpdateHeartbeat{
-                 host_id: ^agent_id,
-                 heartbeat: :critical
-               } = command
-
-        :ok
+      fn
+        command, _ ->
```

Comment:
Shouldn't we test that we are adding the correlation?
Otherwise, do we even test that the correlation_id is added somewhere?